qcfractal/storage_sockets/sqlalchemy_socket.py

```diff
@@ -1959,28 +1973,28 @@ def add_user(self,
         ----------
         username : str
             New user's username
-        password : str
-            The user's password
+        password : str, optional
+            The user's password. If None, a new password will be generated.
         permissions : list of str, optional
             The associated permissions of a user ['read', 'write', 'compute', 'queue', 'admin']
-
+        overwrite: bool, optional
+            Overwrite the user if it already exists.
         Returns
         -------
         tuple
-            Successful insert or not
+            A tuple of (success flag, password)
         """
-        valid_permissions = {'read', 'write', 'compute', 'queue', 'admin'}
 
         # Make sure permissions are valid
-        if not valid_permissions >= set(permissions):
-            raise KeyError("Permissions settings not understood: {}".format(set(permissions) - valid_permissions))
+        if not SQLAlchemySocket.VALID_PERMISSIONS >= set(permissions):
```

Comment:
We typically use `_valid_permissions` for these variables. All caps are used for truly global vars.

Reply:
The all caps is intended to denote a constant:

https://www.python.org/dev/peps/pep-0008/#constants

rather than a class-private variable.

Reply:
I guess it could also be a private constant i.e. `_VALID_PERMISSIONS`

Reply:
renamed to `_valid_permissions` and made into a frozenset